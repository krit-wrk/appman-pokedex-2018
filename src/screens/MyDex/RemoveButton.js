import React from "react";
import { css } from "aphrodite";
import style from "./style";

export const RemoveButton = ({ onClick }) => (
	<div className={css(style.removeButton)} onClick={onClick}>
		X
	</div>
);
