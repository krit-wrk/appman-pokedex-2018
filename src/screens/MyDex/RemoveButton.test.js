import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { RemoveButton } from "./RemoveButton";
import { StyleSheetTestUtils } from "aphrodite";
describe("RemoveButton", () => {
	beforeEach(() => {
		StyleSheetTestUtils.suppressStyleInjection();
	});
	afterEach(() => {
		StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
	});
	it("renders without crashing", () => {
		const fn = jest.fn();
		const { container, getByText } = render(<RemoveButton onClick={fn} />);
		expect(container).toMatchSnapshot();
		fireEvent.click(getByText("X"));
		expect(fn).toHaveBeenCalled();
	});
});
