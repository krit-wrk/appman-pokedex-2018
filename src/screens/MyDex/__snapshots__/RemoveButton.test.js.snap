// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveButton renders without crashing 1`] = `
<div>
  <div
    class="removeButton_5gyw2"
  >
    X
  </div>
</div>
`;
